collection_mutation.cc

```diff
@@ -242,49 +238,56 @@ collection_type_impl::merge(collection_mutation_view a, collection_mutation_view
             // FIXME: should we consider TTLs too?
         };
     };
-    combine(aa.cells.begin(), std::remove_if(aa.cells.begin(), aa.cells.end(), cell_killed(bb.tomb)),
-            bb.cells.begin(), std::remove_if(bb.cells.begin(), bb.cells.end(), cell_killed(aa.tomb)),
+
+    collection_mutation_view_description merged;
+    merged.cells.reserve(a_view.cells.size() + b_view.cells.size());
+
+    combine(a_view.cells.begin(), std::remove_if(a_view.cells.begin(), a_view.cells.end(), cell_killed(b_view.tomb)),
+            b_view.cells.begin(), std::remove_if(b_view.cells.begin(), b_view.cells.end(), cell_killed(a_view.tomb)),
             std::back_inserter(merged.cells),
             compare,
             merge);
-    merged.tomb = std::max(aa.tomb, bb.tomb);
-    return serialize_mutation_form(std::move(merged));
-  });
- });
+    merged.tomb = std::max(a_view.tomb, b_view.tomb);
+
+    return merged.serialize(*this);
+
+    });});
 }
 
 collection_mutation
 collection_type_impl::difference(collection_mutation_view a, collection_mutation_view b) const
 {
- return a.data.with_linearized([&] (bytes_view a_in) {
-  return b.data.with_linearized([&] (bytes_view b_in) {
-    auto aa = deserialize_mutation_form(a_in);
-    auto bb = deserialize_mutation_form(b_in);
+    return a.with_deserialized(*this, [&] (collection_mutation_view_description a_view) {
+    return b.with_deserialized(*this, [&] (collection_mutation_view_description b_view) {
+
```

Comment:
Please unbreak the indent

Reply:
I liked it that way actually, but OK

Reply:
has it changed?

Reply:
Ok, one of the commits had it wrong, another commit changed it later.
I can change it in all the affected commits.